### ISSUE TITLE
PVRTC buffer sizes are computed incorrectly

### ISSUE THREAD

**Issue by zeux** [BinomialLLC/basis_universal#30]
Per PVRTC extension spec https://www.khronos.org/registry/OpenGL/extensions/IMG/IMG_texture_compression_pvrtc.txt:

>     6) How is the imageSize argument calculated for the CompressedTexImage2D
>       and CompressedTexSubImage2D functions. 
>
>       Resolution: For PVRTC 4BPP formats the imageSize is calculated as:
>          ( max(width, 8) * max(height, 8) * 4 + 7) / 8

Width and height are clamped to 8 before computing the imageSize. However, transcoder computes a smaller size - instead of 32 bytes for 4x4 image, it assumes it's just one 8-byte block and returns 8 - which leads to glCompressedTexImage2D generating an error on iOS:

> INVALID_VALUE: compressedTexImage2D: length of ArrayBufferView is not correct for dimensions.

I'm not sure if this is *just* padding - that is, if the workaround of transcoding the first 8 bytes in a 32-byte buffer and passing the resulting buffer to GL is correct, or if the remaining bytes need to be filled with meaningful information.

See https://github.com/mrdoob/three.js/issues/16524#issuecomment-498925744

**Comment by richgel999** [comment]
Thanks - I will fix this ASAP. Thank you very much!

**Comment by richgel999** [comment]
Yes, the transcoder only writes the first block for a 1x1 slice, so the rest of the memory will need to be initialized to something for safety/correctness.

**Comment by richgel999** [comment]
I've checked in fixes for this.
Now transcode_image_level() will always memset() any extra "padding" bytes beyond the blocks we know the transcoder will write to, for PVRTC1.
Also, I updated the wrapper's getImageTranscodedSizeInBytes() and transcodeImage() functions so it computes the correctly padding sizes. I haven't had a chance to test these changes on my iPhone yet, but I'm hoping this is fixed.
